pkg/solana/txm/pendingtx.go

```diff
@@ -81,21 +95,18 @@ type pendingTxContext struct {
 
 func newPendingTxContext() *pendingTxContext {
 	return &pendingTxContext{
-		cancelBy: map[string]context.CancelFunc{},
-		sigToID:  map[solana.Signature]string{},
+		sigToRetryCtx: map[solana.Signature]retryCtx{},
+		sigToTxInfo:   map[solana.Signature]txInfo{},
 
 		broadcastedProcessedTxs: map[string]pendingTx{},
 		confirmedTxs:            map[string]pendingTx{},
 		finalizedErroredTxs:     map[string]finishedTx{},
 	}
 }
 
-func (c *pendingTxContext) New(tx pendingTx, sig solana.Signature, cancel context.CancelFunc) error {
+// New adds a new tranasction in Broadcasted state to the storage
+func (c *pendingTxContext) New(tx pendingTx) error {
```

Comment:
Previously we were adding the first signature here and created some difficulties to restart the retry/bumping cycle when handling the reorg. Removed that part so New is only in charge of creating the new pendingTx